scripts/evaluation_script.r

```diff
@@ -0,0 +1,56 @@
+rm(list = ls())
+# read all the R files under R folder #
+R_path <- "R/"
+purrr::map(paste0(R_path, list.files("R/")), source)
+
+# load observed data #
+
+# Load 2022 NIS data for USA
+nis_usa_2022 <- get_uptake_data(
+    "https://data.cdc.gov/api/views/akkj-j5ru/rows.csv?accessType=DOWNLOAD",
+    state = "Geography",
+    date = c("Time.Period", "Year"),
+    cumulative = "Estimate....",
+    state_key = "data/USA_Name_Key.csv",
+    date_format = "%m/%d/%Y",
+    start_date = "9/2/2022",
+    filters = list(c(
+        "Indicator.Category",
+        "Received updated bivalent booster dose (among adults who completed primary series)"
+    ))
+)
+
+# Load 2023 NIS data for USA
+nis_usa_2023 <- get_uptake_data(
+    "data/NIS_2023-24.csv",
+    state = "geography",
+    date = "date",
+    cumulative = "estimate",
+    state_key = "data/USA_Name_Key.csv",
+    date_format = "%m/%d/%Y",
+    start_date = "9/12/2023",
+    filters = list(c("time_type", "Weekly"), c("group_name", "Overall"))
+)
+
+
+# generate projections initiated at different time #
+all_data <- rbind(nis_usa_2022, nis_usa_2023)
```

Comment:
I agree that this data input is not practical, I did this is to have more data to train. This may cause the issue of 'drop_first' too. Maybe it's worth to put a warning/error sign when the data of multiple seasons is used in python, as in most cases we will only project within one season.

Reply:
I think we will often train the model based on multiple seasons of past data, but we will only project within one season. So maybe your sliding start date for the projection evaluations should only slide within the most recent season represented in `all_data` and never reach back to past seasons. Do you think that is a reasonable approach?

Reply:
I have the sense that Ed's model can only be trained on one season at a time, ie there is no pooling across seasons, because there is no hierarchy? (Or, if it *does* have hierarchy, we shouldn't be starting from there; we should be starting from models that treat each season independently.)

Reply:
No hierarchy at the moment - I would just recommend we keep that possibility in mind for the future